cmd/kubeadm/app/phases/kubelet/flags.go

```diff
@@ -65,6 +66,13 @@ func WriteKubeletDynamicEnvFile(cfg *kubeadmapi.ClusterConfiguration, nodeReg *k
 		pauseImage:               images.GetPauseImage(cfg),
 		registerTaintsUsingFlags: registerTaintsUsingFlags,
 	}
+	criSocket := flagOpts.nodeRegOpts.CRISocket
+	if features.Enabled(cfg.FeatureGates, features.NodeLocalCRISocket) {
+		flagOpts.nodeRegOpts.CRISocket = ""
+		defer func() {
+			flagOpts.nodeRegOpts.CRISocket = criSocket
+		}()
+	}
```

Comment:
I did it this way because it's simple enough and doesn't require passing the `ClusterConfiguration` to `buildKubeletArgs` -> `buildKubeletArgsCommon`. However, if you think it would be better to pass the `ClusterConfiguration` or a bool value indicating whether the feature gate is enabled to the function, I will modify it according to your suggestion

Reply:
To keep the current way, a simple comment may help for code reading.

Reply:
Added.